executor/filter.c

```diff
@@ -0,0 +1,113 @@
+#include "filter.h"
+
+/* Gamestates:
+ *   poison ivy: Motor velocities reversed for 10 seconds
+ *   dehydraiton: Motor velocities stopped for 10 seconds
+ *   hypothermia: Motor velocities slowed until specified otherwise
+ */
+
+// Microseconds between each time we poll shared memory for the current game state
+#define GAMESTATE_POLL_INTERVAL (0.5 * 1000000)
+// Duration of POISON_IVY and DEHYDRATION in milliseconds
+#define DEBUFF_DURATION 10000
+// How much to slow motor velocities when HYPOTHERMIA is ACTIVE
+#define SLOW_SCALAR 0.2
+
+// Some "hardcoded" values to decrease overhead
+// device_name_to_type("KoalaBear")
+#define KOALABEAR 6
```

Comment:
Shouldn't do this cus hardcoding is bad, instead just call the function in the beginning of `gamestate_handler`